autogen.pl: patch generated configure file for ifort support on OS X

ggouaillardet opened the pull request:
Thanks Tien Quang, NGUYEN for sharing the fix at
https://ntq1982.github.io/files/20200621.html

Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>

(cherry picked from commita00530f91f5fd3816094739b8afe8e9150dc1c2b)

jsquyres commented:
Hey @ggouaillardet Were the cherry_pick_check failure notices on this PR legit? Or did they incorrectly flag your commit as not having a cherry-pick line?  I just want to make sure that the CI test is not malfunctioning...

ggouaillardet commented:
The first one was legit: I manually (and incorrectly changed to `open-mpi-mpi@id`)

I fixed it to `open-mpi/ompi@id` but it seems that was not enough ... so I changed it again to `id` and the problem was fixed.

jsquyres commented:
> The first one was legit: I manually (and incorrectly changed to `open-mpi-mpi@id`)
> 
> I fixed it to `open-mpi/ompi@id` but it seems that was not enough ... so I changed it again to `id` and the problem was fixed.

Ok, yeah, that's expected behavior.  It's looking for a naked git commit hash -- not any kind of github identifiers.
